### PR TITLE
WOR-1777 Handle remaining instances of missing azure resources

### PR DESCRIPTION
**Pull request by blakery**
Fixes two instances of `ManagementException`s that were not being handled:
* Exceptions encountered by LandingZoneService when retrieving resources
* Exceptions being handled in a shared helper class

### CODE REVIEW

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/RemoveManagedIdentitiesAzureVmStep.java [hunk 1]
```diff
@@ -33,8 +34,17 @@ public StepResult deleteResource(FlightContext context) {
             .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
 
     ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
-    return AzureVmHelper.removeAllUserAssignedManagedIdentitiesFromVm(
-        azureCloudContext, computeManager, resource.getVmName());
+    var helperResult =
```
Comment (blakery) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/RemoveManagedIdentitiesAzureVmStep.java:37]: I considered some other options, such as refactoring the helper to return something besides a `StepResult`, inspecting failed `StepResult`s in the error handling in the super-class, or adding a generic existence check for resources. 
They each had their merits, but would all require significantly more planning and verification than I was prepared to do as part of this work.

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/DeleteAzureDatabaseStepUnitTest.java [hunk 2]
```diff
@@ -53,7 +56,7 @@ public class DeleteAzureDatabaseStepUnitTest {
   @BeforeEach
   void setup() {
     when(context.getWorkingMap()).thenReturn(workingMap);
-    when(postgresManager.databases()).thenReturn(databases);
+    Mockito.lenient().when(postgresManager.databases()).thenReturn(databases);
```
Comment (blakery) [inline service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/DeleteAzureDatabaseStepUnitTest.java:59]: The new test doesn't get this far.

**Review by marctalbott — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/DeleteAzureControlledResourceStep.java [hunk 3]
```diff
@@ -70,6 +72,18 @@ protected StepResult handleResourceDeleteException(Exception e, FlightContext co
...
       return new StepResult(AzureManagementExceptionUtils.maybeRetryStatus(ex), ex);
     }
+
+    // When retrieving shared resources from the landing zone,
+    // LZS can encounter the same management exceptions as above
+    // To detect this, we need to examine the message for those error codes
+    if (e instanceof LandingZoneServiceApiException && e.getCause() instanceof ApiException) {
+      var msg = e.getCause().getMessage();
+      if (msg != null
+          && missingResourceManagementCodes.stream().anyMatch(code -> msg.contains(code))) {
```
Comment (aherbst-broad) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/DeleteAzureControlledResourceStep.java:82]: Bummer that we're having to string match here. Does that mean there's potential follow-on work to make the LZS graceful in these situations? If so can you drop a ticket?

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/RemoveManagedIdentitiesAzureVmStepUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@Tag("unit")
+@ExtendWith(MockitoExtension.class)
+public class RemoveManagedIdentitiesAzureVmStepUnitTest {
+
+  @Test
+  public void removeManagedIdentitiesAzureFromVm_returnsSuccessOnMissingMRG()
```
Comment (aherbst-broad) [inline service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/RemoveManagedIdentitiesAzureVmStepUnitTest.java:28]: Thanks for starting some coverage on this step.

**Review by aherbst-broad — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/DeleteAzureControlledResourceStep.java (hunk 3 above)
Comment (blakery) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/DeleteAzureControlledResourceStep.java:82]: Yeah, we could probably return a more meaningful error.

**Pull request merged**